File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const categorized = {};
    const N = clothes.length;
    const categories = [];
    for (let i = 0; i < N; i++){
        const pair = clothes[i];
        const category = pair[1];
        const cloth = pair[0];
        if (!categorized[category]){
            categorized[category] = 0;
        }
        categorized[category]++;
        if (!categories.includes(category))
        categories.push(category);
    }
    let answer = 1;
    for (category of categories){
        answer = answer * (categorized[category] + 1);
    }
    return answer - 1;
    
}

File: Python3/프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    hash_map = {}
    for cloth in clothes:
        name = cloth[0]
        category = cloth[1]
        if category in hash_map:
            hash_map[category] += 1
        else:
            hash_map[category] = 1
    
    answer = 1
    for count in hash_map.values():
        answer *= (count + 1)
    return answer - 1